bundler/lib/bundler/cli/exec.rb

```diff
@@ -77,6 +81,17 @@ def ruby_shebang?(file)
         "#!#{Gem.ruby}\n",
       ]
 
+      if Gem.win_platform? # remove `bat` suffix
+        script_file = File.join(File.dirname(file), File.basename(file, File.extname(file)))
+        return false unless Pathname(script_file).exist?
+        if File.zero?(script_file)
+          Bundler.ui.warn "#{script_file} is empty"
+          return false
+        end
+        first_line = File.open(file, "r") {|f| f.read(5) }
+        return first_line.start_with?("@ECHO")
```

Comment:
I'd say the "shebang equivalent" here lies in the last line of the `.bat` file, where the path to `ruby.exe` is actually included?